mcpserver/mcpserver.go

```diff
@@ -0,0 +1,107 @@
+package mcpserver
+
+import (
+	"context"
+
+	"dagger.io/dagger"
+	"github.com/dagger/container-use/environment"
+	"github.com/dagger/container-use/repository"
+	"github.com/mark3labs/mcp-go/mcp"
+)
+
+func dagFromContext(ctx context.Context) *dagger.Client {
+	dag, ok := ctx.Value(daggerClientKey{}).(*dagger.Client)
```

Comment:
🐛 **Bug (Critical):** dagFromContext panics when dagger client is not found in context. This should return an error instead of panicking to prevent runtime crashes.

```suggestion
func dagFromContext(ctx context.Context) *dagger.Client {
	dag, ok := ctx.Value(daggerClientKey{}).(*dagger.Client)
	if !ok {
		return nil
	}
	return dag
}
```